tests/python/unittest/test_gluon_rnn.py

```diff
@@ -890,152 +657,53 @@ def __init__(self, rnn_size, time_step, **kwargs):
-
-    args, outs, auxs = outputs.infer_shape(rnn_t0_data=in_shape,
-                                           rnn_t1_data=in_shape,
-                                           rnn_t2_data=in_shape)
-    assert outs == [out_shape] * 3
-
-
 def check_rnn_forward(layer, inputs):
     inputs.attach_grad()
     layer.initialize()
     with mx.autograd.record():
         layer.unroll(3, inputs, merge_outputs=True)[0].backward()
         mx.autograd.backward(layer.unroll(3, inputs, merge_outputs=False)[0])
-    mx.nd.waitall()
+mx.npx.waitall()
 
 
 def test_rnn_cells():
     check_rnn_forward(gluon.rnn.Conv1DLSTMCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
     check_rnn_forward(gluon.rnn.Conv1DRNNCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
     check_rnn_forward(gluon.rnn.Conv1DGRUCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
 
     net = mx.gluon.rnn.SequentialRNNCell()
     net.add(gluon.rnn.Conv1DLSTMCell((5, 7), 10, (3,), (3,)))
     net.add(gluon.rnn.Conv1DRNNCell((10, 5), 11, (3,), (3,)))
     net.add(gluon.rnn.Conv1DGRUCell((11, 3), 12, (3,), (3,)))
-    check_rnn_forward(net, mx.nd.ones((8, 3, 5, 7)))
-
-
-def test_convrnn():
-    cell = gluon.rnn.Conv1DRNNCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DRNNCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DRNNCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_convlstm():
-    cell = gluon.rnn.Conv1DLSTMCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DLSTMCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DLSTMCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_convgru():
-    cell = gluon.rnn.Conv1DGRUCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DGRUCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DGRUCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_conv_fill_shape():
-    cell = gluon.rnn.Conv1DLSTMCell((0, 7), 10, (3,), (3,))
-    cell.hybridize()
-    check_rnn_forward(cell, mx.nd.ones((8, 3, 5, 7)))
-    assert cell.i2h_weight.shape[1] == 5, cell.i2h_weight.shape[1]
-
-
-def test_lstmp():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead?